posts/deploy-coder-on-fly-io/index.md

```diff
@@ -0,0 +1,106 @@
+# Deploy Coder on Fly.io
+
+Fly.io is a platform for deploying applications to the edge. It's a great fit for Coder because it's easy to deploy and scale, and it's easy to manage. In this guide, we'll walk through deploying Coder on Fly.io.
+
+## Prerequisites
+
+- A Fly.io [account](https://fly.io/signup).
+- [Fly CLI](https://fly.io/docs/getting-started/installing-flyctl/) installed on your machine.
+
+## Deploy Coder
+
+1. Log in to Fly.io with the CLI:
+
+```bash
+flyctl auth login
+```
+
+2. Create a new fly postgres database:
+
+```bash
+flyctl postgres create DATABASE-NAME
+```
+
+3. Create a new fly app:
+
+```bash
+flyctl apps create APP-NAME
+```
+
+4. Connect to the database with the coder fly app:
+
+```bash
+flyctl postgres connect DATABASE-NAME --app APP-NAME
+```
+
+This will print out a connection string. Copy the connection string and save it to be used in the next step.
+
+5. Edit the `fly.toml` file and update as per the example below:
+
+```toml
+app = "APP-NAME"
+kill_signal = "SIGINT"
+kill_timeout = 5
+primary_region = "ams"  # See a list of regions here: https://fly.io/docs/reference/regions/
+
+[experimental]
+  auto_rollback = true
+  private_network = true  # This is needed for Coder to connect to the database
+
+[build]
+   image = "ghcr.io/coder/coder:latest"
+
+[env]
+  CODER_ACCESS_URL = "https://APP-NAME.fly.dev" # make sure to replace APP-NAME with your app name
+  CODER_HTTP_ADDRESS = "0.0.0.0:3000"
+  CODER_PG_CONNECTION_URL = "ADD YOUR POSTGRES CONNECTION STRING HERE"
```

Comment:
Similarily, I think Fly adds an environment variable for the DB URL when you "attach" it. Can we reference that?

Reply:
same for `DATABASE_URL`

Reply:
For postgress database, `flyctl` adds a secret to the app with name `DATABSE_URL` that we can directly use as an environment variable somehow the assignment `CODER_PG_CONNECTION_URL=$DATABASE_URL` assignment does not work.
> It seems like interpolating environment variables within fly.toml is not supported.

We can either,
1. Copy the connection string to fly.toml
2. Create a new secret `CODER_PG_CONNECTION_URL` with the connection string.